common/pose.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
```

Comment:
I think could be kinda overkill - not really sure what sort of agent wouldn't have all 4...

Reply:
Yeah I agree. I have been thinking about this and what you said below about using templates for ``AntAgent`` and ``ObjectData`` etc. and I'm thinking that a better strategy would be to make the "getter" functions templates rather than having these templated superclasses. Then we just use duck-typing for our agent classes.